File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/EmployeeAddress.ts
export class IEmployeeAddress {
    EmployeeName: string;
    EmployeeAddressID: number;
    EmployeeID: number;
    AddressTypeID: number;
    Address1: string;
    Address2: string;
    City: string;
    State: string;
    Country: string;
    ZipCode: string;
    StartDate: Date;
    EndDate: Date;
    CreatedBy: string;
    CreatedDate: Date;
    ModifiedBy: string;
    ModifiedDate: Date;
    TimeStamp: string;
}
export class IEmployeeAddressDisplay {
    employeeAddressID: number;
    employeeID: number;
    employeeName: string;
    addressTypeID: number;
    address1: string;
    address2: string;
    city: string;
    state: string;
    country: string;
    zipCode: string;
    startDate: Date;
    endDate: Date;
}

File: ILT.IHR.DTO/EmployeeAddress.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace ILT.IHR.DTO
{
    public class EmployeeAddress : AbstractDataObject
    {
        public int EmployeeAddressID { get; set; }
        public int EmployeeID { get; set; }
        public int AddressTypeID { get; set; }
        [Required]
        public string Address1 { get; set; }
        public string Address2 { get; set; }
        [Required]
        public string City { get; set; }
        [Required]
        public string State { get; set; }
        [Required]
        public string Country { get; set; }
        [Required]
        public string ZipCode { get; set; }
        [Required]
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
            
    }
}
